patroni/ctl.py

```diff
@@ -1404,6 +1416,151 @@ def _do_failover_or_switchover(action: str, cluster_name: str, group: Optional[i
+    :param group: filter Citus group within we should perform a failover or switchover. If ``None``, user will be
+        prompted for filling it -- unless *force* is ``True``, in which case an exception is raised.
+    :param switchover_leader: name of the leader site passed as switchover option.
+    :param candidate: name of a standby site to be promoted.
+    :param force: perform the switchover without asking for confirmations.
+    :param scheduled: timestamp when the switchover should be scheduled to occur. If ``now`` perform immediately.
+
+    :raises:
+        :class:`PatroniCtlException`: if:
+            * Patroni is running on a Citus cluster, but no *group* was specified; or
+            * a switchover was requested but the cluster has no leader site (or multisite is not active); or
+            * *switchover_leader* does not match the current leader site of the cluster; or
+            * cluster has no candidate sites available for the operation; or
+            * current leader and *candidate* are the same; or
+            * *candidate* is not a site of the cluster; or
+            * trying to schedule a switchover in a cluster that is in maintenance mode; or
+            * user aborts the operation.
+    """
+    dcs = get_dcs(cluster_name, group)
+    cluster = dcs.get_cluster()
+    click.echo('Current cluster topology')
+    output_members(cluster, cluster_name, group=group)
+
+    if is_citus_cluster() and group is None:
+        if force:
+            raise PatroniCtlException('For Citus clusters the --group must me specified')
+        else:
+            group = click.prompt('Citus group', type=int)
+            dcs = get_dcs(cluster_name, group)
+            cluster = dcs.get_cluster()
+
+    config = global_config.from_cluster(cluster)
+
+    cluster_leader = cluster.leader and cluster.leader.name
+
+    if not cluster_leader:
+        raise PatroniCtlException('This cluster has no leader')
+
+    if cluster.leader and cluster.leader.multisite:
+        leader_site = (cluster.leader.multisite.get('name') if not cluster.leader.multisite.get('standby_config') else
+                       cluster.leader.multisite.get('standby_config', {}).get('leader_site'))
+    else:
+        raise PatroniCtlException('Multisite is not active or there is no leader site, cannot switch sites')
+
+    if switchover_leader is None:
+        if force:
+            switchover_leader = leader_site
+        else:
+            prompt = 'Leader site'
+            switchover_leader = click.prompt(prompt, type=str, default=leader_site)
+
+    if leader_site != switchover_leader:
+        raise PatroniCtlException(f'Site {switchover_leader} is not the leader of cluster {cluster_name}')
+
+    # multisite_dcs = get_dcs(cluster_name, group, True)
+    # multisite_cluster = multisite_dcs.get_cluster()
+
+    candidate_names = [str(m.multisite['name']) for m in cluster.members
+                       if m.multisite and m.multisite['name'] != leader_site]
```

Comment:
I think this should not work, the other sites members are not listed in this sites DCS. Meaning candidate_names will only contain the site name that this is executed on. This is only correct when executed on standby site in a 2 node cluster.